core/src/main/java/hudson/cli/BuildCommand.java

```diff
@@ -135,14 +142,15 @@ protected int run() throws Exception {
         }
 
         if (checkSCM) {
-            if (job.poll(new StreamTaskListener(stdout, getClientCharset())).change == Change.NONE) {
+            SCMTriggerItem item = SCMTriggerItem.SCMTriggerItems.asSCMTriggerItem(job);
+            if (item != null && !item.poll(new StreamTaskListener(stdout, getClientCharset())).hasChanges()) {
```

Comment:
If it is null, we cannot check SCM. Maybe makes sense to fail the command in such case